### PR TITLE
Add Passio Nutrition AI Food Search Tool to Community Package

### PR DESCRIPTION
**Pull request by ivyas21**
### Description
We propose adding a new tool to the `community` package, enabling integration with Passio Nutrition AI for food search functionality. This tool will provide a simple interface for retrieving nutrition facts through the Passio Nutrition AI API, simplifying user access to nutrition data based on food search queries.

### Implementation Details
- **Class Structure:** Implement `NutritionAI`, extending `BaseTool`. It includes an `_run` method that accepts a query string and, optionally, a `CallbackManagerForToolRun`.
- **API Integration:** Use `NutritionAIAPI` for the API wrapper, encapsulating all interactions with the Passio Nutrition AI and providing a clean API interface.
- **Error Handling:** Implement comprehensive error handling for API request failures.

### Expected Outcome
- **User Benefits:** Enable easy querying of nutrition facts from Passio Nutrition AI, enhancing the utility of the `langchain_community` package for nutrition-related projects.
- **Functionality:** Provide a straightforward method for integrating nutrition information retrieval into users' applications.

### Dependencies
- `langchain_core` for base tooling support
- `pydantic` for data validation and settings management
- Consider `requests` or another HTTP client library if not covered by `NutritionAIAPI`.

### Tests and Documentation
- **Unit Tests:** Include tests that mock network interactions to ensure tool reliability without external API dependency.
- **Documentation:** Create an example notebook in `docs/docs/integrations/tools/passio_nutrition_ai.ipynb` showing usage, setup, and example queries.

### Contribution Guidelines Compliance
- Adhere to the project's linting and formatting standards (`make format`, `make lint`, `make test`).
- Ensure compliance with LangChain's contribution guidelines, particularly around dependency management and package modifications.

### Additional Notes
- Aim for the tool to be a lightweight, focused addition, not introducing significant new dependencies or complexity.
- Potential future enhancements could include caching for common queries to improve performance.

### Twitter Handle
-  Here is our Passio AI [twitter handle](https://twitter.com/@passio_ai) where we announce our products.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Comment by ivyas21** [comment]
@efriis @eyurtsev  Could you please look at this PR?

**Review by hwchase17 — COMMENTED** [review]
need to run format - `make format`

**Comment by ivyas21** [comment]
@hwchase17  I'm fixing it.

**Comment by ivyas21** [comment]
@hwchase17 I ran `make format`

File: libs/community/langchain_community/utilities/passio_nutrition_ai.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+"""Util that invokes the Passio Nutrition AI API.
+"""
+from datetime import datetime, timedelta
+from typing import Union, Dict, List, final
+
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (baskaryan) [inline libs/community/langchain_community/utilities/passio_nutrition_ai.py:10]: can we remove this? dotenv isn't a langchain-community dependency

Comment (ivyas21) [reply libs/community/langchain_community/utilities/passio_nutrition_ai.py:10]: Sure.

Comment (ivyas21) [reply libs/community/langchain_community/utilities/passio_nutrition_ai.py]: Fixed it and push the changes.

**Comment by ivyas21** [comment]
@baskaryan I removed print and unused library also updated with `pydantic` . Thank you!!

**Review by baskaryan — APPROVED** [review]

**Comment by ivyas21** [comment]
@baskaryan , clean up some lint issue.

**Comment by ivyas21** [comment]
@baskaryan , I fixed some remaining issue. Hopefully it gets all clear from CI/CD. Thank you.

**Comment by baskaryan** [comment]
fyi you can run 
```bash
cd libs/community
poetry install --with test,lint,typing
make format
make lint
```
to run linter locally and see any issues

**Comment by ivyas21** [comment]
@baskaryan , Is it merged to master?

**Comment by ivyas21** [comment]
@baskaryan  No issue on locally. All clear. but here it says : Some checks haven’t completed yet
59 successful and 1 pending checks. Not sure which one is pending checks?  Thank you.

**Pull request merged**